ethcore/src/engines/mod.rs

```diff
@@ -393,7 +397,7 @@ pub trait Engine<M: Machine>: Sync + Send {
 	fn step(&self) {}
 
 	/// Stops any services that the may hold the Engine and makes it safe to drop.
-	fn stop(&self) {}
+	fn stop(&mut self) {}
```

Comment:
is this really needed?

Reply:
It use to be needed, when we use IOService,  but then I switched to an dedicated thread, this is nolonger needed, but the old interface was simply not enought in most case I think.